exp/api/v1alpha3/azuremachinepool_types.go

```diff
@@ -165,6 +170,16 @@ type (
 	}
 )
 
+// GetConditions returns the list of conditions for an AzureMachinePool API object.
+func (amp *AzureMachinePool) GetConditions() clusterv1.Conditions {
+	return amp.Status.Conditions
+}
+
+// SetConditions will set the given conditions on an AzureMachinePool object
+func (amp *AzureMachinePool) SetConditions(conditions clusterv1.Conditions) {
```

Comment:
Should this change also include defining basic conditions for AzureMachinePools and setting them in the controller like https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/714/files#diff-825ec7ff34f98855f2347c7f7769438dR284 did?

Reply:
Sounds reasonable. I'll look into it.

Reply:
@nprokopic are you still planning to do this? If not, we can merge this as-is and mark the issue as incomplete

Reply:
Yes, I do, just couldn't find the time in the previous weeks.

Can we merge this and have a separate pull request for setting conditions?

Lately I was working on conditions implementation for Giant Swarm's azure-operator (where we use CAPI and CAPZ types), here's what we added for current alpha release (so not fully completed yet) for Cluster, AzureCluster and MachinePool https://github.com/giantswarm/azure-operator/pull/1112, and here is current WIP for AzureMachinePool https://github.com/giantswarm/azure-operator/pull/1123.
Once we wrap these up, I would take a look at what's currently done in CAPZ, and then open an issue with suggestions for conditions that would make sense for CAPZ.

Reply:
Okay let's do that.